src/main/java/io/kuzzle/sdk/core/KuzzleDocument.java

```diff
@@ -196,11 +191,13 @@ public KuzzleDocument refresh(final KuzzleOptions options, final KuzzleResponseL
       this.kuzzle.query(this.dataCollection.makeQueryArgs("read", "get"), content, options, new OnQueryDoneListener() {
         @Override
         public void onSuccess(JSONObject args) {
+          KuzzleDocument newDocument = new KuzzleDocument(KuzzleDocument.this.dataCollection);
+
           try {
-            KuzzleDocument.this.put("body", args.getJSONObject("_source"));
-            KuzzleDocument.this.put("_version", args.get("_version"));
+            newDocument.put("body", args.getJSONObject("_source"));
+            newDocument.put("_version", args.get("_version"));
             if (listener != null) {
```

Comment:
When a check is added to enforce a non-null `listener` argument, this test will always return `true`. So it should be removed along with the addition of the new check.